singlestoredb/management/job.py

```diff
@@ -0,0 +1,332 @@
+                Dictionary of values
+
+            Returns
+            -------
+            :class:`TargetConfig`
+
+            """
+            out = cls(
+                database_name=obj.get('databaseName'),
+                resume_target=bool(obj.get('resumeTarget')),
+                target_id=obj.get('targetID'),
+                target_type=obj.get('targetType'),
+            )
+
+            return out
+
+
+class Job(object):
+    """
+    Scheduled Notebook Job definition.
+
+    This object is not directly instantiated. It is used in results
+    of API calls on the :class:`JobsManager`. See :meth:`JobsManager.run`.
+    """
+
+    job_id: str
+    name: str
+    description: str
+    project_id: str
+    created_at: datetime.datetime
+    enqueued_by: str
+    completed_executions_count: int
+    terminated_at: datetime.datetime
+    job_metadata : List[ExecutionMetadata]
+    executionConfig: ExecutionConfig
+    schedule: Schedule
+    targetConfig: Optional[TargetConfig]
+
+    def __init__(
+        self,
+        job_id: str,
+        name: str,
+        description: str,
+        project_id: str,
+        created_at: datetime.datetime,
+        enqueued_by: str,
+        completed_executions_count: int,
+        terminated_at: datetime.datetime,
+        job_metadata: List[ExecutionMetadata],
+        executionConfig: ExecutionConfig,
+        schedule: Schedule,
+        targetConfig: Optional[TargetConfig],
+    ):
+        self.job_id = job_id
+        self.name = name
+        self.description = description
+        self.project_id = project_id
+        self.created_at = created_at
+        self.enqueued_by = enqueued_by
+        self.completed_executions_count = completed_executions_count
+        self.terminated_at = terminated_at
+        self.job_metadata = job_metadata
+        self.executionConfig = executionConfig
+        self.schedule = schedule
+        self.targetConfig = targetConfig
+
+    @classmethod
+    def from_dict(cls, obj: Dict[str, Any], manager: Manager) -> 'Job':
+        """
+        Construct a Job from a dictionary of values.
+
+        Parameters
+        ----------
+        obj : dict
+            Dictionary of values
+
+        Returns
+        -------
+        :class:`Job`
+
+        """
+
+        out = cls(
+            job_id=obj['jobID'],
+            name=obj['name'],
+            description=obj['description'],
+            project_id=obj['projectID'],
+            created_at=to_datetime(obj['createdAt']),
+            enqueued_by=obj['enqueuedBy'],
+            completed_executions_count=int(obj['completedExecutionsCount']),
+            terminated_at=to_datetime(obj['terminatedAt']),
+            job_metadata=[ExecutionMetadata.from_dict(x) for x in obj['jobMetadata']],
+            executionConfig=ExecutionConfig.from_dict(obj['executionConfig']),
+            schedule=Schedule.from_dict(obj['schedule']),
+            targetConfig=TargetConfig.from_dict(obj.get('targetConfig'))
```

Comment:
This will need to be `TargetConfig.from_dict(obj.get('targetConfig')) if 'targetConfig' in obj else None`

Reply:
Also, `targetConfig` should be `target_config`.